src/util/Button.ts

```diff
@@ -0,0 +1,47 @@
+import Graphics from "../Graphics";
+import Input from "../Input";
+import { getMousePos, getTexture, input } from "../main.js";
+import Subtexture from "../texture/Subtexture";
+import Texture from "../texture/Texture";
+
+let widgetsTex: Texture;
+let buttonTex: Subtexture;
+let hoverButtonTex: Subtexture;
+let clickButtonTex: Subtexture;
+
+export class Button {
+	x: number; y: number; w: number; h: number; text: string
+
+	constructor(x: number, y: number, w: number, h: number, text: string) {
+		this.x = x
+		this.y = y
+		this.w = w
+		this.h = h
+		this.text = text
+	}
+
+	static loadTexture() {
+		widgetsTex = getTexture("tiny/textures/gui/widgets.png")
+		buttonTex = widgetsTex.getSubtexture(0, 66, 200, 20);
+		hoverButtonTex = widgetsTex.getSubtexture(0, 86, 200, 20);
+		clickButtonTex = widgetsTex.getSubtexture(0, 86, 200, 20);
+	}
+
+	static touchingRect(x: number, y: number, w: number, h: number) {
+		let mouse = getMousePos();
```

Comment:
use `input.mouse`, `getMousePos()` is in block coordinates (it's off by `game.width/2` due to translate)